Example switcher

jaketrent opened the pull request:
- Wanted to author in markdown since this is the common case for code authoring in docs
- Chose code block in markdown because it allows an opening and closing symbol
-  Chose a --- / hr symbol to split examples
- Downside to markdown is lack of rich data structure. Best idea so far: in a following PR, parse jsdocs for each example to parse out title and description

danethurber commented on docs/src/components/mdx/code-block.tsx:
```diff
@@ -1,22 +1,80 @@
+  const options = isSwitcher
+    ? Array.from(Array(codes.length)).map((_, i) => ({
+        value: i,
+        label: 'Example #' + i,
```
obviously a way to define a title will be nice to add down the rpad

danethurber commented on docs/src/components/mdx/code-block.tsx:
```diff
@@ -1,22 +1,80 @@
+      }))
+    : [{ value: 0, label: 'Single example' }]
+
+  const [selectedOption, setSelectedOption] = React.useState(options[0].value)
```
i'm not getting the initially selected option in the dropdown value

![Screen Shot 2020-09-15 at 9 37 24 AM](https://user-images.githubusercontent.com/226356/93232380-14a76c80-f737-11ea-8bb7-63a437b1284d.png)

danethurber commented on docs/src/pages/index.mdx:
````diff
@@ -56,18 +28,35 @@ const Example: React.FC = () => {
-```
 
-```typescript codesandbox=default
+---
+
+/*
+ * For now, the description can live in a code block.
+ * 
+ * If we don't put this in markdown, we will be authoring in .tsx instead of .mdx.
+ * 
+ * Later, we can update code-block.tsx to extract these, parse markdown and display as html
+ */
````
maybe we write in frontmatter style and parse it out with something like https://github.com/docs/frontmatter

danethurber reviewed (approved):
love the approach. couple comments. nice work!

jaketrent commented:
Frontmatter was a great idea. I was going to use a kind of jsdoc. Fixed the selected options too.  There may be a bug or 2 in the dropdown. If the value is just a number, the item label gets followed by a 0 and the label is not shown in the dropdown when selected. Fixed that by prepending 'value' to the item values.
